lib/active_merchant/billing/gateway.rb

```diff
@@ -200,6 +200,16 @@ def supports_network_tokenization?
         false
       end
 
+      def add_options_to_post_if_present(post, options, selected_options)
+        selected_options.each do |option|
+          add_option_to_post_if_present(post, options, option)
+        end
+      end
+
+      def add_option_to_post_if_present(post, options, option)
+        post[option] = options[option] if options[option]
+      end
```

Comment:
🤔 Hrmmmmmnamingishard. I can't tell if these reservations are worth worrying about but: the term `option` is more general than a field with a value that would be added to a request, and this also sounds a bit like it takes the entire options hash in the context of the transaction, when it really only handles a passed subset. I think they're fine as argument names though, but `field` or similar might be better for the method name...

Reply:
Updated.

Reply:
Looks great!